src/app/app-initializers.ts

```diff
@@ -35,18 +36,19 @@ export const appInitializers = {
         const url = location.href.replace(location.origin, "");
         router.navigateByUrl(url, { skipLocationChange: true });
       });
-
       // update the user context for remote error logging and tracking and load config initially
-      loginState.subscribe((newState) => {
-        if (newState === LoginState.LOGGED_IN) {
-          const username = currentUser.value.name;
-          LoggingService.setLoggingContextUser(username);
-          analyticsService.setUser(username);
-        } else {
-          LoggingService.setLoggingContextUser(undefined);
-          analyticsService.setUser(undefined);
-        }
-      });
+      combineLatest([loginState, currentlyLoggedIn]).subscribe(
+        ([newState, user]) => {
+          if (newState === LoginState.LOGGED_IN) {
+            const username = user?.getId(true);
+            LoggingService.setLoggingContextUser(username);
+            analyticsService.setUser(username);
```

Comment:
it might still make sense to have the username (auth/keycloak layer) here for logging, rather than the entity (business logic layer). But I think it doesn't hurt much to have it this way also.